src/providers/arcgisrest/qgsafsshareddata.cpp

```diff
@@ -42,6 +42,33 @@ long long QgsAfsSharedData::featureCount() const
   return mObjectIds.size() - mDeletedFeatureIds.size();
 }
 
+QgsRectangle QgsAfsSharedData::extent() const
+{
+  if ( mDataSource.sql().isEmpty() )
+    return mExtent;
+
+  return QgsArcGisRestQueryUtils::getExtent( mDataSource.param( QStringLiteral( "url" ) ), mDataSource.sql(), mDataSource.authConfigId(),
```

Comment:
Should we not cache this value?

You could have a QgsRectangle mSubsetExtent which defaults to an empty value and assign the getExtent() value to it. Whenever a new subset string is set, clear mSubsetExtent?

Reply:
@nirvn there's already caching of the extent handled at the vector layer level -- let's avoid the complexity of a double cache :laughing:

Reply:
@nyalldawson , I was walking up the tree, never thinking of looking down for caching, doh! :)